server/eventlog/eventlog.go

```diff
@@ -320,6 +305,35 @@ type WriteCloserWithContext interface {
 type EventLogWriter struct {
 	WriteCloserWithContext
 	chunkstoreWriter *chunkstore.ChunkstoreWriter
+	lastChunk        *elpb.LiveEventLogChunk
+	keyValueStore    interfaces.KeyValStore
+	eventLogPath     string
+}
+
+func (w *EventLogWriter) writeChunkToKeyValStore(ctx context.Context, writeRequest *chunkstore.WriteRequest, writeResult *chunkstore.WriteResult, chunk []byte, volatileTail []byte) {
+	if writeResult.Close {
+		keyval.SetProto(ctx, w.keyValueStore, w.eventLogPath, nil)
+		return
+	}
+	chunkId := chunkstore.ChunkIndexAsStringId(writeResult.LastChunkIndex + 1)
+	if chunkId == chunkstore.ChunkIndexAsStringId(math.MaxUint16) {
+		keyval.SetProto(ctx, w.keyValueStore, w.eventLogPath, nil)
+		return
+	}
+	curChunk := &elpb.LiveEventLogChunk{
+		ChunkId: chunkId,
+		Buffer:  append(chunk, volatileTail...),
+	}
+
+	if !proto.Equal(w.lastChunk, curChunk) {
```

Comment:
uber nit: I think it would be clearer to early return if they are equal (like you do above), then unnest the keyval.SetProto, but up to you.

Reply:
Done.